### PR TITLE
Fixed a bug where the scheduler would get None

### PR DESCRIPTION
**Pull request by wassimseif**
Fixed a bug where the scheduler would get None for T_max if user is passing num_epochs instead of num_train_steps.

Without this fix the user would get an exception

### CODE REVIEW

**Comment by lucidrains** [comment]
thanks!
